python/py-sentence-transformers/Portfile

```diff
@@ -0,0 +1,92 @@
+                    sentence embeddings that can be used in\
+                    unsupervised scenarios: Semantic textual\
+                    similarity via cosine-similarity, clustering,\
+                    semantic search. We provide an increasing number\
+                    of state-of-the-art pretrained models that can be\
+                    used to derive sentence embeddings. See Pretrained\
+                    Models. Details of the implemented approaches can\
+                    be found in our publication: Sentence-BERT:\
+                    Sentence Embeddings using Siamese BERT-Networks\
+                    (published at EMNLP 2019). You can use this code\
+                    to easily train your own sentence embeddings, that\
+                    are tuned for your specific task. We provide\
+                    various dataset readers and you can tune sentence\
+                    embeddings with different loss function, depending\
+                    on the structure of your dataset. For further\
+                    details, see Train your own Sentence Embeddings.
+
+checksums           rmd160  1b0b7cdb92bd9c5466bd05f57979a1954604bae3 \
+                    sha256  88cd9df7ba045b1fd68a36a6ee82548dfc4fa57f47947c489a3cd639429557f3 \
+                    size    46900
+
+python.versions     36 37
+
+if {${name} ne ${subport}} {
+    depends_build-append \
+                    port:py${python.version}-setuptools
+
+    depends_run-append \
+                    port:py${python.version}-nltk \
+                    port:py${python.version}-numpy \
+                    port:py${python.version}-pytorch \
+                    port:py${python.version}-scikit-learn \
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-transformers
+
+    # Fix sentence-transformers's dependecies on deprecated names pytorch-transformers
+    # remove this after the upstream has been fixed
+    post-extract {
+        # pytorch-transformers
+        reinplace -E \
+            {s/^([[:space:]]*(pytorch-transformers)[[:space:]]*)==([[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*)?/transformers>=\3/g} \
+            ${worksrcpath}/requirements.txt
+        reinplace -E \
+            {s/^([[:space:]]*['"])(pytorch-transformers)([[:space:]]*)==([[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*['"],)?/\1transformers\3>=\4/g} \
+            ${worksrcpath}/setup.py
+        fs-traverse f ${worksrcpath} {
+            if {[file isfile ${f}]
+                && [string match "*.py" ${f}]} {
+                reinplace -E -q \
+                    {s/((import|from)[[:space:]]+)(pytorch_transformers)(\.[[:alnum:]_]+)*/\1transformers/g} \
+                    ${f}
+                reinplace -E -q \
+                    {s/(pytorch_(transformers\.))/\2/g} \
+                    ${f}
+                reinplace -q \
+                    {s/BertAdam/AdamW/g} \
+                    ${f}
+            }
+        }
+    }
```

Comment:
This block should be unnecessary in a few days. See https://github.com/UKPLab/sentence-transformers/issues/72#issuecomment-562491436

Reply:
Upstream updated now, and this block deleted.